libs/server/Storage/Session/ObjectStore/SetOps.cs

```diff
@@ -361,6 +362,33 @@ public unsafe GarnetStatus SetScan<TObjectContext>(ArgSlice key, long cursor, st
 
         }
 
+        /// <summary>
+        /// Returns the members of the set resulting from the union of all the given sets.
+        /// Keys that do not exist are considered to be empty sets.
+        /// </summary>
+        /// <param name="keys"></param>
+        /// <param name="output"></param>
+        /// <param name="objectStoreContext"></param>
+        /// <typeparam name="TObjectContext"></typeparam>
+        /// <returns></returns>
+        public GarnetStatus SetUnion<TObjectContext>(ArgSlice[] keys, out HashSet<byte[]> output, ref TObjectContext objectStoreContext)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, SpanByte, GarnetObjectStoreOutput, long>
+        {
+            output = new HashSet<byte[]>(new ByteArrayComparer());
+
+            foreach (var key in keys)
+            {
+                if (GET(key.ToArray(), out var currObject, ref objectStoreContext) == GarnetStatus.OK)
+                {
+                    var currSet = ((SetObject)currObject.garnetObject).Set;
```

Comment:
You don't interact directly with the set object, you need to go through the storage layer to do that. Please refer to other commands that interact with more than one object (for example: LMOVE, ZDIFF)